web/src/beta/features/ProjectSettings/innerPages/PublicSettings/AliasSettings/index.tsx

```diff
@@ -5,38 +5,51 @@ import {
 } from "@reearth/beta/features/Editor/Publish/PublishToolsPanel/common";
 import { Button, Icon } from "@reearth/beta/lib/reearth-ui";
 import CommonField from "@reearth/beta/ui/fields/CommonField";
+import { useProjectFetcher } from "@reearth/services/api";
 import { config } from "@reearth/services/config";
 import { useT } from "@reearth/services/i18n";
 import { useNotification } from "@reearth/services/state";
 import { styled, useTheme } from "@reearth/services/theme";
 import { FC, useCallback, useMemo, useState } from "react";
 
 import { PublicAliasSettingsType } from "..";
+import {
+  SettingsProjectWithTypename,
+  StoryWithTypename
+} from "../PublicSettingsDetail";
 
 import EditPanel from "./EditPanel";
 
 export type AliasSettingProps = {
-  alias: string;
   isStory?: boolean;
-  defaultAlias?: string;
+  alias?: string;
+  settingsItem?: SettingsProjectWithTypename | StoryWithTypename;
   onUpdateAlias?: (settings: PublicAliasSettingsType) => void;
   onClose?: () => void;
   onSubmit?: (alias?: string) => void;
 };
 const AliasSetting: FC<AliasSettingProps> = ({
-  alias,
   isStory,
-  defaultAlias,
+  settingsItem,
   onUpdateAlias
 }) => {
   const theme = useTheme();
   const t = useT();
+  const { useProjectAliasCheckQuery } = useProjectFetcher();
```

Comment:
I think i know why we used lazy query earlier.
We should not implement using `useProjectAliasCheckQuery` since it will be triggered everytime load this page, instead we need a `checkAlias` query directy. We can implement one on projectApi, and use it when user want to update alias or reset alias.

Reply:
For example
```
const checkAlias = useCallback(async (alias?: string) => {
    if (!alias) return null;

    const [fetchCheckAlias] = useLazyQuery(CHECK_PROJECT_ALIAS);
    const { data, error } = await fetchCheckAlias({
      variables: { alias }
    });

    if (error) {
      console.error("Failed to check alias:", error);
      return null;
    }

    return data?.checkProjectAlias;
  }, []);
  ```